### ISSUE TITLE
Crash when using PublishSingleFile

### ISSUE THREAD

**Issue by i3ym** [ppy/osu-framework#4504]
When publishing an application using PublishSingleFile=true it crashes when trying to load osu framework.

```
> dotnet publish -c Release -r linux-x64 /p:PublishSingleFile=true
> ./bin/Release/net5.0/linux-x64/publish/test
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.IO.Path.Combine(String path1, String path2)
   at osu.Framework.Platform.DesktopGameHost.GetDefaultGameStorage()
   at osu.Framework.Platform.GameHost.Run(Game game)
```

Looks like a problem is in the DebugUtils.GetEntryPath method which gets Assembly.Location and it [does not work](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/il3000) in single-file apps, at least in net5

**Comment by bdach** [comment]
While this failure is probably the most blocking one, I'm not super sure we're able to fully support single-file without significant development effort, which I don't think is likely to happen anytime soon. Aside from the failure above, the following probably need to be checked:

https://github.com/ppy/osu-framework/blob/07945da4c031a76e0fea87184ca721f4a3d1d146/osu.Framework/Graphics/Video/VideoDecoder.cs#L547
https://github.com/ppy/osu-framework/blob/07945da4c031a76e0fea87184ca721f4a3d1d146/osu.Framework/IO/Stores/DllResourceStore.cs#L20
https://github.com/ppy/osu-framework/blob/99786238a02e0d6b69da86dd52e5506ee8ec0566/osu.Framework/RuntimeInfo.cs#L19-L28
https://github.com/ppy/osu-framework/blob/41f8306e8788a79ab5e05c609bad2da66856e4dd/osu.Framework/Platform/GameHost.cs#L217-L219

**Comment by peppy** [comment]
Wiling to say that if you're trying to use this feature it'd be appreciated if you can fix/submit any required changes to make it work, because it's out of scope for us in the short term.

**Comment by jai-x** [comment]
Closed by https://github.com/ppy/osu-framework/pull/4653 I think?

**Comment by frenzibyte** [comment]
Yep, any issues can be reported separately, the one mentioned in the OP should be resolved by #4653.
